commonui/src/main/java/com/hyperwallet/android/ui/common/insight/HyperwalletInsight.java

```diff
@@ -216,21 +216,28 @@ public Handler getHandler() {
      * @param context      Context where the tracking is happening
      * @param pageName     an arbituary pageName the user is currently viewing as defined by the app
      * @param pageGroup    an arbituary pageGroup the user is currently viewing as defined by the app
+     * @param params    additional information that can be added to make the tracking event more useful
      * @param errorInfoMap contains information regarding the error
      */
     public void trackError(@NonNull final Context context, @NonNull final String pageName,
-            @NonNull final String pageGroup, @NonNull final Map<String, String> errorInfoMap) {
+            @NonNull final String pageGroup, @Nullable final Map<String, String> params,
+            @NonNull final Map<String, String> errorInfoMap) {
```

Comment:
I think, here we are confusing more. We can send as ErrorInfo from where the event occurs and also here we can get the same as per confluence. Then we can process the same EventProcessor.java in inside project. Those params also describes the error in detail. So, i think params also part of error info. 
Current scenario, We need to send the **"null"** value HashMap as a argument if params are not required.
Other options, I think recommended way to call a method with different number of argument by using Method Overloading.
    public void trackError(@NonNull final Context context, @NonNull final String pageName,
            @NonNull final String pageGroup, @NonNull final Map<String, String> errorInfoMap) {
trackError(context, pageName,pageGroup, null, errorInfoMap);// instead of sending everywhere as a null
}
    public void trackError(@NonNull final Context context, @NonNull final String pageName,
            @NonNull final String pageGroup, @Nullable final Map<String, String> params,
            @NonNull final Map<String, String> errorInfoMap) {
//Actual code
}
<img width="891" alt="Screenshot 2019-11-20 at 8 23 26 PM" src="https://user-images.githubusercontent.com/52773997/69251748-e53bd800-0bd7-11ea-9157-1de69031c266.png">
<img width="1666" alt="Screenshot 2019-11-20 at 8 43 14 PM" src="https://user-images.githubusercontent.com/52773997/69251796-f4228a80-0bd7-11ea-8e23-82bc3455cb6a.png">
<img width="1395" alt="Screenshot 2019-11-20 at 8 57 26 PM" src="https://user-images.githubusercontent.com/52773997/69252138-904c9180-0bd8-11ea-9c76-7b3565a331a4.png">

Reply:
I created `add()` and `addAll()` methods to ErrorParamsBuilder. So the user can optionally add the extra params. This will avoid the need to pass null.